Document Alert: GOV.UK Account SLOs - Error rates (#3147)
alphagov/govuk-developer-docs: source/manual/alerts/slo-rate-checks-account.html.md

---

> owner_slack: "#govuk-account-tech"

Reviewer: `#govuk-accounts-tech` - not `#govuk-account-tech`

> title: GOV.UK Account SLO error rate exceeded 1% in 10 minute window

Reviewer: 1% is part of the SLO, if we change the SLO then we don't really want to have to update the title of this page too. 
 Also, it's good if this title is close to the Icinga alert titles, so it's easier to find in search.

How about:

```
SLO Error Rate on GOV.UK Accounts has exceeded an acceptable threshold across the last 10 minutes
```

section: Icinga alerts
layout: manual_layout
parent: "/manual.html"
---

The GOV.UK Account team has defined Service Level Objectives (SLOs) for
reliability of the `account-api`.

You can read more about these in [GOV.UK Accounts Service Level
Objectives](https://docs.google.com/document/d/1HVXlyy2vdy9U0O-_7Nw8_Oso7VXQeKgJ2_-bi4pKlz0/edit)

> ## Warning - "HTTP Error on GOV.UK Account has exceeded an acceptable threshold across the last 10 minutes"

Reviewer: I don't think the `Warning - "..."` is adding anything here, just use `...` directly

This warning triggers when over 1% of all HTTP status codes are deemed "bad
responses" as a proportion of all responses within a rolling 10 minute time
period.

This is an early warning that, if these errors continue at a consistent rate,

> the GOV.UK Account team would spend its entire error budget of 1% across the

Reviewer: Here (and below) there's no need to mention the 1% again, just say

```
the GOV.UK Account team would spend its entire error budget across the
```

> full 4 week monitoring period.

Reviewer: Can probably drop the length of the monitoring period too.

This warning automatically triggers a message to the `#govuk-accounts-tech`
slack channel.

No further action is required from GOV.UK 2ndline or any other team.

## Warning - "Latency on GOV.UK Account has exceeded an acceptable threshold across the last 10 minutes"

This warning triggers when over 1% of all HTTP responses are deemed

> "insufficiently quick" as a proportion of all responses within a rolling 10

Reviewer: "too slow" is better than "insufficiently quick"

Reviewer 2: good point @barrucadu, i thought "insufficiently quick" was the term used in Icinga. Whatever term we use in the docs should reflect what the user sees.

---
owner_slack: "#govuk-account-tech"
title: GOV.UK Account SLO error rate exceeded 1% in 10 minute window

> section: Icinga alerts

Reviewer 3: Is this an Icinga alert, or do they stem from Prometheus?

Author: So... here's the config:
https://github.com/alphagov/govuk-puppet/pull/11171

This is my first venture into this stuff, but I'm pretty sure we're not touching prometheus.
My understanding is that it'd got:
Graphite  query -> Icinga warning?

Reviewer: Yeah, these are Icinga  alerts

layout: manual_layout
parent: "/manual.html"
---

> The GOV.UK Accounts team has defined Service Level Objectives (SLOs) for

Reviewer 2: I think it's GOV.UK Account team rather than GOV.UK Account**s** team.

reliability of the `account-api`.

> You can read more about these in [GOV.UK Accounts Service Level

Reviewer 2: I think it's GOV.UK Account team rather than GOV.UK Account**s** team.

Objectives](https://docs.google.com/document/d/1HVXlyy2vdy9U0O-_7Nw8_Oso7VXQeKgJ2_-bi4pKlz0/edit)

> ## Warning - "HTTP Error on GOV.UK Accounts has exceeded an acceptable threshold across the last 10 minutes"

Reviewer 2: I think it's GOV.UK Account team rather than GOV.UK Account**s** team.

This triggers when, within a rolling 10 minute window, over 1% of all HTTP
status codes were deemed "bad responses", as a proprotion of all responsed in

> that period.

Reviewer 2: Suggest following change: "This warning triggers when over 1% of all HTTP status codes are deemed "bad responses" as a proportion of all responses within a rolling 10 minute time period."

This is intended as an early warning that, if these errors continued at a

> consistent rate, the GOV.UK Accounts team is on route to spend it's entire

Reviewer 3: ```suggestion
consistent rate, the GOV.UK Accounts team is en route to spend its entire
```

Author: I think Jon helped me with some better wordingfor this bit, thanks!

> error budget of 1% across the full 4 week monitoring period.

Reviewer 2: Suggest following change: "This is an early warning that, if these errors continue at a consistent rate, the GOV.UK Account team would spend its entire error budget of 1% across the full 4 week monitoring period."

This warning automatically triggers a message to the `#govuk-accounts-tech`
slack channel.

No further action is required from GOV.UK 2ndline or any other team.

> ## Warning - "Latency on GOV.UK Accounts has exceeded an acceptable threshold across the last 10 minutes"

Reviewer 2: I think it's GOV.UK Account team rather than GOV.UK Account**s** team.

This triggers when, within a rolling 10 minute window, over 1% of all HTTP
responses occur with insufficently quickly, as a proprotion of all responsed in

> that period.

Reviewer 2: See change in previous section.

I think there's a word missing or an extra word in line 33: "...responses occur with insufficiently quickly..."

This is intended as an early warning that, if these errors continued at a
consistent rate, the GOV.UK Accounts team is on route to spend it's entire

> error budget of 1% across the full 4 week monitoring period.

Reviewer 2: See change in previous section.
